Description: Make lorder work in current Debian. 
Author: Michael Meskes <meskes@debian.org>

--- bsdmainutils/usr.bin/lorder/lorder.sh.orig	2020-05-28 11:41:36.165253077 +0200
+++ bsdmainutils/usr.bin/lorder/lorder.sh	2020-05-28 11:41:36.165253077 +0200
@@ -45,12 +45,12 @@
 esac
 
 # temporary files
-R=$(mktemp -t _reference_)
-S=$(mktemp -t _symbol_)
+R=$(mktemp -t _reference_XXXXXX)
+S=$(mktemp -t _symbol_XXXXXX)
 NM=${NM:-nm}
 
 # remove temporary files on HUP, INT, QUIT, PIPE, TERM
-trap "rm -f $R $S $T; exit 1" 1 2 3 13 15
+trap "rm -f $R $S $T; exit 1" HUP INT QUIT PIPE TERM
 
 # make sure all the files get into the output
 for i in $*; do
@@ -80,7 +80,7 @@
 if [ $(expr "$*" : '.*\.a[[:>:]]') -ne 0 ]; then
 	sort -u -o $S $S
 	sort -u -o $R $R
-	T=$(mktemp -t _temp_)
+	T=$(mktemp -t _temp_XXXXXX)
 	comm -23 $R $S >$T
 	mv $T $R
 fi
